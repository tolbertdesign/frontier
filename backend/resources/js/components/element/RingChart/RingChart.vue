
<template>
  <svg
    viewBox="0 0 36 36"
    class="circular-chart md:my-2 mx-auto"
  >
    <path
      class="circle-bg"
      d="M18 2.0845
                a 15.9155 15.9155 0 0 1 0 31.831
                a 15.9155 15.9155 0 0 1 0 -31.831"
    />
    <path
      :stroke-dasharray="`${value}, 100`"
      class="circle"
      d="M18 2.0845
                a 15.9155 15.9155 0 0 1 0 31.831
                a 15.9155 15.9155 0 0 1 0 -31.831"
    />
    <text
      x="18"
      y="20.35"
      class="percentage"
    >{{ value }}%</text>
  </svg>
</template>

<script>
export default {
  name: 'RingChart',
  props: {
    value: {
      type: Number,
      default: 0,
    },
  },
}
</script>
