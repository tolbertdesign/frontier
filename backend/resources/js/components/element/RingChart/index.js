import RingChart from './RingChart.vue'

export default RingChart
