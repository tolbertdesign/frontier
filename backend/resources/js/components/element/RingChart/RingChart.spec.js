import { shallowMount, createLocalVue } from '@vue/test-utils'
import RingChart from './RingChart.vue'

const localVue = createLocalVue()

const value = 10

describe('RingChart', () => {
  it('Snapshop test for RingChart', () => {
    const wrapper = shallowMount(RingChart, {
      localVue,
      propsData: {
        value: value,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
