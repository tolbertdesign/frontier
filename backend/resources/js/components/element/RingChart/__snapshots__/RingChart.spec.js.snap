// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RingChart Snapshop test for RingChart 1`] = `
<svg viewBox="0 0 36 36" class="circular-chart md:my-2 mx-auto">
  <path d="M18 2.0845
            a 15.9155 15.9155 0 0 1 0 31.831
            a 15.9155 15.9155 0 0 1 0 -31.831" class="circle-bg"></path>
  <path stroke-dasharray="10, 100" d="M18 2.0845
            a 15.9155 15.9155 0 0 1 0 31.831
            a 15.9155 15.9155 0 0 1 0 -31.831" class="circle"></path> <text x="18" y="20.35" class="percentage">10%</text>
</svg>
`;
